### PR TITLE
Rebuild MapboxNavigationNotification for each update

### LINKED ISSUE #1188: MapboxNavigationNotification#updateNotificationViews TransactionTooLargeException
Run into the following 💥 while testing

```
java.lang.RuntimeException: android.os.TransactionTooLargeException: data parcel size 524216 bytes
  at android.app.NotificationManager.notifyAsUser(NotificationManager.java:408)
  at android.app.NotificationManager.notify(NotificationManager.java:370)
  at android.app.NotificationManager.notify(NotificationManager.java:346)
  at com.mapbox.services.android.navigation.v5.navigation.MapboxNavigationNotification.updateNotificationViews(MapboxNavigationNotification.java:150)
  at com.mapbox.services.android.navigation.v5.navigation.MapboxNavigationNotification.updateNotification(MapboxNavigationNotification.java:72)
  at com.mapbox.services.android.navigation.v5.navigation.NavigationNotificationProvider.updateNavigationNotification(NavigationNotificationProvider.java:23)
  at com.mapbox.services.android.navigation.v5.navigation.RouteProcessorThreadListener.onNewRouteProgress(RouteProcessorThreadListener.java:33)
  at com.mapbox.services.android.navigation.v5.navigation.RouteProcessorHandlerCallback$1.run(RouteProcessorHandlerCallback.java:97)
  at android.os.Handler.handleCallback(Handler.java:873)
  at android.os.Handler.dispatchMessage(Handler.java:99)
  at android.os.Looper.loop(Looper.java:193)
  at android.app.ActivityThread.main(ActivityThread.java:6669)
  at java.lang.reflect.Method.invoke(Native Method)
  at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:493)
  at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:858)
Caused by: android.os.TransactionTooLargeException: data parcel size 524216 bytes
  at android.os.BinderProxy.transactNative(Native Method)
  at android.os.BinderProxy.transact(Binder.java:1127)
  at android.app.INotificationManager$Stub$Proxy.enqueueNotificationWithTag(INotificationManager.java:1558)
  at android.app.NotificationManager.notifyAsUser(NotificationManager.java:405)
  at android.app.NotificationManager.notify(NotificationManager.java:370) 
  at android.app.NotificationManager.notify(NotificationManager.java:346) 
  at com.mapbox.services.android.navigation.v5.navigation.MapboxNavigationNotification.updateNotificationViews(MapboxNavigationNotification.java:150) 
  at com.mapbox.services.android.navigation.v5.navigation.MapboxNavigationNotification.updateNotification(MapboxNavigationNotification.java:72) 
  at com.mapbox.services.android.navigation.v5.navigation.NavigationNotificationProvider.updateNavigationNotification(NavigationNotificationProvider.java:23) 
  at com.mapbox.services.android.navigation.v5.navigation.RouteProcessorThreadListener.onNewRouteProgress(RouteProcessorThreadListener.java:33) 
  at com.mapbox.services.android.navigation.v5.navigation.RouteProcessorHandlerCallback$1.run(RouteProcessorHandlerCallback.java:97) 
  at android.os.Handler.handleCallback(Handler.java:873) 
  at android.os.Handler.dispatchMessage(Handler.java:99) 
  at android.os.Looper.loop(Looper.java:193) 
  at android.app.ActivityThread.main(ActivityThread.java:6669) 
  at java.lang.reflect.Method.invoke(Native Method) 
  at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:493) 
  at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:858) 
```

### PR DESCRIPTION
**Pull request by danesfeder**
Fixes #1188 and Closes #1441 

See #1441 for further explanation / discussion.

### CODE REVIEW

File: libandroid-navigation/src/main/java/com/mapbox/services/android/navigation/v5/navigation/MapboxNavigationNotification.java [hunk 1]
```diff
@@ -48,6 +47,8 @@
   private int currentManeuverId;
   private boolean isTwentyFourHourFormat;
   private String etaFormat;
+  private final Context context;
```
Comment (Guardiola31337) [inline libandroid-navigation/src/main/java/com/mapbox/services/android/navigation/v5/navigation/MapboxNavigationNotification.java]: Minor comment - to note that should be an application context, what about calling it `applicationContext` instead? As in https://github.com/mapbox/mapbox-navigation-android/blob/5eabb77d9b5cd061fccf96d3f6f49ce1bc5501c1/libandroid-navigation/src/main/java/com/mapbox/services/android/navigation/v5/navigation/MapboxNavigationNotification.java#L50

**Comment by Danny-James** [comment]
@danesfeder The data in the Notification is still not matching the `BannerText` how can I force an update of the Notification?

And is their also a way to prevent the Notification from showing at all?

screenshot_20181106-105038_route info
screenshot_20181106-105042_route info

**Comment by danesfeder** [comment]
@Danny-James you can create your own notification with https://www.mapbox.com/android-docs/navigation/overview/navigation-options/#creating-a-custom-notification

**Comment by Danny-James** [comment]
@danesfeder Instead of creating my own, is their a documented way of stopping the default notification, as i don't think we need it to be honest?

But if we do in the future at least i'm aware on disabling the default and/or creating my own.

**Comment by danesfeder** [comment]
@Danny-James we had to remove the option to disable because on Oreo+ a `ForegroundService` must be accompanied with a notification due to it being the most aggressive service you can run on Android.  We do this so navigation doesn't get killed by the system in the background.

**Comment by Danny-James** [comment]
@danesfeder does the default notification have a layout file I could replicate?

**Pull request merged**